src/workerd/api/node/util.c++

```diff
@@ -240,4 +241,39 @@ jsg::JsValue UtilModule::getBuiltinModule(jsg::Lock& js, kj::String specifier) {
   return js.undefined();
 }
 
+void UtilModule::processExitImpl(jsg::Lock& js, int code) {
+  if (IoContext::hasCurrent()) {
+    // There are a few things happening here. First, we abort the current IoContext
+    // in order to shut down this specific request....
+    auto message =
+        kj::str("The Node.js process.exit(", code, ") API was called. Canceling the request.");
+    auto& ioContext = IoContext::current();
+    // If we have a tail worker, let's report the error.
+    KJ_IF_SOME(tracer, ioContext.getWorkerTracer()) {
+      auto ex = KJ_ASSERT_NONNULL(js.error(message).tryCast<jsg::JsObject>());
+      tracer.addException(ioContext.now(), ex.get(js, "name"_kj).toString(js),
+          ex.get(js, "message"_kj).toString(js), ex.get(js, "stack"_kj).toString(js));
+      ioContext.abort(js.exceptionToKj(ex));
+    } else {
+      ioContext.abort(JSG_KJ_EXCEPTION(FAILED, Error, kj::mv(message)));
+    }
+    // ...then we tell the isolate to terminate the current JavaScript execution.
+    // Oddly however, this does not appear to *actually* terminate the thread of
+    // execution unless we trigger the Isolate to handle the intercepts, which
+    // calling v8::JSON::Stringify does. Weird... but ok? As long as it works
+    // TODO(soon): Investigate if there is a better approach to triggering the
+    // interrupt handling.
+    js.v8Isolate->TerminateExecution();
+    v8::JSON::Stringify(js.v8Context(), js.str()).IsEmpty();
```

Comment:
How did you come up with JSON stringification as the best solution here?

I mean, I guess it's fine, just curious what else you tried.

Reply:
Picking through the v8 code to see what paths would trigger handling the scheduled interrupts with the least amount of overhead. The JSON stringify method handles the interrupts very early in the function and is super easy to call so it presented the easiest option here. Tho I admit it's rather funky.